mrcasals/town_simulator
# Feature/modernize project

Thanks for sharing this project! I've had a lot of fun experimenting with various simulation options :smile:

I wanted to run this locally so I made a quick update to modernize the project setup:

## This replaces the previous setup with...

- `pnpm` for rapid and efficient package management.
- `vite` as a dev server and bundler.
- `vitest` to run existing tests with 100% passing tests.
- `c8` to generate tests coverage reports.
- `prettier` to keep the project consistently formatted. The `pnpm format`script will take care of everything. Also another tip is to enable the Prettier plugin for your editor to take care of formatting when saving, to be able to focus fully on the logic of the code rather than the style.
- Also updated the README with new instructions.

This project is now also prepared for typescript conversion and made a quick attempt at that too, although it is not as straightforward with  the `immutable` package. However, everything is prepared to start changing files into `.ts` (maybe starting with `src/index.ts` and changing the import in `index.html` to use `./src/index.ts` instead of `./src/index.js`).

One final note is that this PR removes some of the old setup with Docker and Circle-CI. Could likely be added back easily if you want it in the future.

# How to test locally:

Either...
1. Follow the updated setup instructions in README.md:

Or do the following...
1. Install https://pnpm.io and run `pnpm install` in the root of the project
2. Run `pnpm dev` and see the running simulation at `http://127.0.0.1:5173`
3. Run the tests with `pnpm test` and `pnpm test:coverage` (which will generate the coverage HTML report that can be viewed by running `pnpm dev` and going to `http://127.0.0.1:5173/coverage/` (note the trailing slash here which is very important))
